integration_tests_server/index.js

```diff
@@ -0,0 +1,176 @@
+  app.get('/v2/send/:jsonp_id', (req, res) => {
+    const channel = decode_base64(req.query.channel);
+    const event = decode_base64(req.query.event);
+    const data = JSON.parse(decode_base64(req.query.data));
+    const socket_id = decode_base64(req.query.socket_id || "");
+
+    pusher.trigger(channel, event, data);
+
+    res.set({
+      'Access-Control-Allow-Origin': '*',
+      'Content-Type': 'text/javascript'
+    });
+    res.send(`Pusher.Integration.ScriptReceivers[${req.params.jsonp_id}](null, {});`)
+  });
+
+  // JSONP echo
+  app.get('/jsonp/echo/:id', (req, res) => {
+    const decoded_params =
+          Object.keys(req.query)
+          .filter(key => !["id", "receiver", "splat", "captures"].includes(key))
+          .reduce((obj, key) => {
+            obj[key] = decode(req.query[key]);
+            return obj;
+          }, {});
+
+    const receiver = params.hasOwnProperty("receiver") ? decode(params["receiver"]) : "Pusher.JSONP.receive";
+
+    res.set({
+      'Content-Type': 'text/javascript; charset=utf-8',
+      'Cache-Control': 'private, max-age=0, must-revalidate'
+    });
+    res.send(`${receiver}(${req.params.id}, null, ${JSON.stringify(decoded_params)});`)
+  });
+
+  // JSONP echo
+  // pusher-js 2.2 JSONP API
+  app.get('/v2/jsonp/echo/:id', (req, res) => {
+    const decoded_params =
+          Object.keys(req.query)
+          .filter(key => !["id", "splat", "captures"].includes(key))
+          .reduce((obj, key) => {
+            obj[key] = decode(req.query[key]);
+            return obj;
+          }, {});
+
+    res.set({
+      'Content-Type': 'text/javascript; charset=utf-8',
+      'Cache-Control': 'private, max-age=0, must-revalidate'
+    });
+    res.send(`Pusher.ScriptReceivers[${req.params.id}](null, ${JSON.stringify(decoded_params)});`)
+  });
+
+  // ScriptRequest echo
+  app.get('/v2/script_request/echo', (req, res) => {
+    const decoded_params =
+          Object.keys(req.query)
+          .filter(key => !["id", "receiver", "splat", "captures"].includes(key))
+          .reduce((obj, key) => {
+            obj[key] = req.query[key];
+            return obj;
+          }, {});
+
+    res.set({
+      'Content-Type': 'text/javascript; charset=utf-8',
+      'Cache-Control': 'private, max-age=0, must-revalidate'
+    });
+    res.send(`${req.query.receiver}(null, ${JSON.stringify(decoded_params)});`);
+  });
+
+  app.get( '/jsonp/500/:id', (req, res) => {
+    res.sendStatus(500);
+  });
+
+  app.get( '/jsonp/404/:id', (req, res) => {
+    res.sendStatus(404);
+  });
+
+  app.listen(port, () => {
+    console.log(`Integration tests auth server listening on port ${port}`);
+  })
+}
+
+start_app(3000, {
+  appId: process.env.INTEGRATION_TESTS_APP_MT1_APP_ID,
+  key: "c49a449b916bb4246729",
+  secret: process.env.INTEGRATION_TESTS_APP_MT1_SECRET,
+  cluster: "mt1",
+});
+
+start_app(3001, {
+  appId: process.env.INTEGRATION_TESTS_APP_EU_APP_ID,
+  key: "cb7d5e50f5669e5c9ae6",
```

Comment:
Let's move keys as well to env vars.

Reply:
They are (and were) already present in the actual integration tests to instantiate the pusher object. We could also use the env var there I guess.